### PR TITLE
Enable TAliEnFile to retry writes (now location based) and mirror the uploaded copy

### PR DESCRIPTION
**Pull request by mimarpe**

### CODE REVIEW

File: net/alien/inc/TAlienFile.h [hunk 1]
```diff
@@ -56,7 +54,7 @@ class TAlienFile : public TXNetFile {
    TAlienFile(const char *purl, Option_t *option = "",
               const char *ftitle = "", Int_t compress = 1,
               Bool_t parallelopen = kFALSE, const char *lurl = 0,
-              const char *authz = 0);
```
Comment (dberzano) [inline net/alien/inc/TAlienFile.h:59]: Use spaces not tabs.

File: net/alien/inc/TAlienFile.h [hunk 2]
```diff
@@ -31,9 +31,7 @@
 //                                                                      //
 //////////////////////////////////////////////////////////////////////////
 
-#ifndef ROOT_TXNetFile
```
Comment (dberzano) [inline net/alien/inc/TAlienFile.h]: Why did you remove the fencing?

File: net/alien/src/TAlienFile.cxx [hunk 3]
```diff
@@ -46,65 +46,66 @@
...
+/// For a description of the option and other arguments see the TFile ctor.
+/// The preferred interface to this constructor is via TFile::Open().
+///
+/// Warning: TAlienFile objects should only be created through the factory functions:
+///    TFile::Open("alien://...");
+/// or
+///    TAlienFile::Open("alien://...");
+///
+/// Don't use "new TAlienFile" directly unless you know, what you are doing
+///
```
Comment (dberzano) [inline net/alien/src/TAlienFile.cxx]: Remove the last empty comment line.

File: net/alien/src/TAlienFile.cxx [hunk 4]
```diff
@@ -260,41 +261,48 @@ TAlienFile *TAlienFile::Open(const char *url, Option_t *option,
...
-   }
-
    TString fALfn = file;
 
    int imagenr = 0;
+   TString exSes = "";
 
-   do {
+   // number of global loops. for read goes one by one, while write returns N(QoS)
+   int maxtries = MAX_FILE_IMAGES_READ;
```
Comment (dberzano) [inline net/alien/src/TAlienFile.cxx:269]: OK this is not an English essay but it's "matrices" not "matrixes" :-)

File: net/alien/src/TAlienFile.cxx [hunk 5]
```diff
@@ -333,190 +341,194 @@ TAlienFile *TAlienFile::Open(const char *url, Option_t *option,
...
+            if (anchor.Length()) {
+               newurl += "#";
+               newurl += anchor;
+               TString lUrlfile = lUrl.GetFile();
+               TString lUrloption;
+               lUrloption = "zip=";
+               lUrloption += anchor;
+               lUrloption += "&mkpath=1";
+               lUrl.SetFile(lUrlfile);
+               // lUrl.SetAnchor(anchor);
```
Comment (dberzano) [inline net/alien/src/TAlienFile.cxx]: Remove the comment.

**Review by dberzano — CHANGES_REQUESTED** [review]
Apart from some style comments, I have to believe you on the rest...

File: net/alien/src/TAlienFile.cxx (hunk 4 above)
Comment (dberzano) [reply net/alien/src/TAlienFile.cxx:269]: Forget it I have misread haha!

File: net/alien/inc/TAlienFile.h (hunk 2 above)
Comment (mimarpe) [reply net/alien/inc/TAlienFile.h]: I think this wasn't on the file in root/master...
I guess it doesn't hurt to add

**Review by dberzano — APPROVED** [review]

**Comment by dberzano** [comment]
Thanks, looks good to me, we'll merge and test on Monday though.

**Comment by dberzano** [comment]
Validation started!

**Comment by dberzano** [comment]
See [ALIROOT-7715](https://alice.its.cern.ch/jira/browse/ALIROOT-7715): we've successfully validated your fix, thanks!
